networkx/drawing/nx_pylab.py

```diff
@@ -470,6 +474,8 @@ def draw_networkx_nodes(
         labelleft=False,
     )
 
+    ax.margins(margins[0], margins[1])
+
```

Comment:
```suggestion
    if margins is not None:
        ax.margins(margins[0], margins[1])

```
I think this is the cause of (at least) most errors.
We have to treat the default case explicitly.